NDTensors/src/lib/GPUArraysCoreExtensions/src/gpuarrayscore.jl

```diff
@@ -13,3 +15,9 @@ function set_storagemode(type::Type, param)
 end
 
 function cpu end
+
+cpu(E::Exposed) = cpu(unexpose(E))
+
+function Base.append!(Ecollection::Exposed{<:AbstractGPUArray}, collections...)
+  return @allowscalar append!(unexpose(Ecollection), collections...)
+end
```

Comment:
Let's define separate versions for each GPU backend so we are prepared once they start getting `append!` implementations.

Reply:
Okay I made an append for each GPU backend and I made a comment about how metal is currently broken and removed the out of date append!! code from the library